src/state/state.ts

```diff
@@ -2,32 +2,44 @@ import type { BigNumber } from 'ethers';
 import { produce, type Draft } from 'immer';
 
 import type { Config } from '../config/schema';
-import type { LocalSignedData, AirnodeAddress, TemplateId } from '../types';
+import type { ChainId, DApiName, DecodedDataFeed, DataFeedId, SignedData } from '../types';
 
 interface GasState {
   gasPrices: { price: BigNumber; timestampMs: number }[];
   sponsorLastUpdateTimestampMs: Record<string, number>;
 }
 
+export interface DataFeedValue {
+  value: BigNumber;
+  timestamp: number;
+}
+
+export interface DapiState {
+  dataFeed: DecodedDataFeed;
+  dataFeedValues: Record<ChainId, DataFeedValue>;
+  updateParameters: Record<ChainId, UpdateParameters>;
+}
+
+export interface UpdateParameters {
+  deviationThresholdInPercentage: BigNumber;
+  deviationReference: BigNumber;
+  heartbeatInterval: number;
+}
+
 export interface State {
   config: Config;
   dataFetcherInterval?: NodeJS.Timeout;
   gasPriceStore: Record<string, Record<string, GasState>>;
-  signedApiStore: Record<AirnodeAddress, Record<TemplateId, LocalSignedData>>;
-  signedApiUrlStore: Record<string, Record<AirnodeAddress, string>>;
+  signedApiStore: Record<DataFeedId, SignedData>;
+  signedApiUrlStore: string[];
+  dapis: Record<DApiName, DapiState>;
 }
 
 type StateUpdater = (draft: Draft<State>) => void;
 
 let state: State | undefined;
 
-export const getState = (): State => {
-  if (!state) {
-    throw new Error('State is undefined.');
-  }
-
-  return state;
-};
+export const getState = (): State => state!;
```

Comment:
Why change this? It seems quite dengarous to me.

Reply:
I just thought it'd make it a bit cleaner. The state being undefined should be impossible (there won't be any config) - but it doesn't bother me, I've reverted the change :)